pm/recipient_test.go

```diff
@@ -0,0 +1,851 @@
+	sender, b, v, ts, faceValue, winProb, sig := newRecipientFixture(t)
+
+	r, err := NewRecipient(b, v, ts, faceValue, winProb)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Config stub validator with valid non-winning tickets
+	v.SetIsValidTicket(true)
+
+	params, err := r.TicketParams(sender)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var wg sync.WaitGroup
+	var errCount uint64
+
+	for i := 0; i < 50; i++ {
+		wg.Add(1)
+
+		go func(senderNonce uint64) {
+			defer wg.Done()
+
+			ticket := newTicket(sender, params, senderNonce)
+
+			_, err := r.ReceiveTicket(ticket, sig, params.Seed)
+			if err != nil {
+				atomic.AddUint64(&errCount, 1)
+			}
+		}(uint64(i))
+	}
+
+	wg.Wait()
+
+	if errCount == 0 {
+		t.Error("expected more than zero senderNonce errors for concurrent ticket receipt")
+	}
+}
+
+func TestRedeemWinningTickets_InvalidSessionID(t *testing.T) {
+	_, b, v, ts, faceValue, winProb, _ := newRecipientFixture(t)
+
+	r, err := NewRecipient(b, v, ts, faceValue, winProb)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Config stub ticket store to fail load
+	ts.loadShouldFail = true
+
+	if err := r.RedeemWinningTickets("foo"); err == nil {
+		t.Error("expected ticket store error")
+	}
+}
+
+func TestRedeemWinningTickets_SingleTicket_GetDepositError(t *testing.T) {
+	sender, b, v, ts, faceValue, winProb, sig := newRecipientFixture(t)
+
+	r, err := NewRecipient(b, v, ts, faceValue, winProb)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	params, err := r.TicketParams(sender)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Config stub validator with valid winning tickets
+	v.SetIsValidTicket(true)
+	v.SetIsWinningTicket(true)
+	// Config stub broker with zero penalty escrow
+	b.SetPenaltyEscrow(sender, big.NewInt(0))
+
+	// Test get deposit error
+	ticket := newTicket(sender, params, 0)
+
+	won, err := r.ReceiveTicket(ticket, sig, params.Seed)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !won {
+		t.Fatal("expected valid winning ticket")
+	}
+
+	err = r.RedeemWinningTickets(ticket.RecipientRandHash.Hex())
+	if err == nil {
+		t.Error("expected get deposit error")
+	}
+}
+
+func TestRedeemWinningTickets_SingleTicket_GetPenaltyEscrowError(t *testing.T) {
+	sender, b, v, ts, faceValue, winProb, sig := newRecipientFixture(t)
+
+	r, err := NewRecipient(b, v, ts, faceValue, winProb)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	params, err := r.TicketParams(sender)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Config stub validator with valid winning tickets
+	v.SetIsValidTicket(true)
+	v.SetIsWinningTicket(true)
+	// Config stub broker with zero deposit
+	b.SetDeposit(sender, big.NewInt(0))
+
+	// Test get penalty escrow error
+	ticket := newTicket(sender, params, 0)
+
+	won, err := r.ReceiveTicket(ticket, sig, params.Seed)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !won {
+		t.Fatal("expected valid winning ticket")
+	}
+
+	err = r.RedeemWinningTickets(ticket.RecipientRandHash.Hex())
+	if err == nil {
```

Comment:
Same question for this test and the one above - is there a way we can make this check more specific to know that we're indeed failing for the reason we expect to be failing?